include/caffe/util/signal_handler.h

```diff
@@ -0,0 +1,24 @@
+#ifndef INCLUDE_CAFFE_UTIL_SIGNAL_HANDLER_H_
+#define INCLUDE_CAFFE_UTIL_SIGNAL_HANDLER_H_
+
+#include "caffe/proto/caffe.pb.h"
+#include "caffe/solver.hpp"
+
+namespace caffe {
+
+class SignalHandler {
+ public:
+  // Contructor. Specify what action to take when a signal is received.
+  SignalHandler(SolverParameter_Action SIGINT_action,
+                SolverParameter_Action SIGHUP_action);
+  ActionCallback GetActionFunction();
+ private:
+  SignalHandler();  // Not implemented.
```

Comment:
>If the default constructor is not declared private, the compiler will provide it implicitly.

C++ does generate a default constructor but **only if you don't provide one of your own**. Since you've already provided one: `SignalHandler(SolverParameter_Action SIGINT_action, SolverParameter_Action SIGHUP_action);`, a default constructor won't be generated by compiler implicitly. So a private constructor declaration here is unnecessary.

You may verify with this simple program:

    class C {
     public:
      C(int) {}
    };
    int main() { C c; } // compilation fails because there's no default constructor in class C.

Anyway, this is only a small issue. The PR is great!